nav2_pure_pursuit_controller/src/pure_pursuit_controller.cpp

```diff
@@ -123,15 +161,15 @@ geometry_msgs::msg::TwistStamped PurePursuitController::computeVelocityCommands(
 
 void PurePursuitController::setPlan(const nav_msgs::msg::Path & path)
 {
-  // Transform global path into the robot's frame
-  global_plan_ = transformGlobalPlan(path);
+  global_plan_ = path;
```

Comment:
Doesn't this need to be transformed into the odometry frame?

Reply:
You removed the transformation function from this

Reply:
Why move it into the compute function?

Reply:
I've moved it to compute function to calculate this every tick. The map to odom and odom to map transform could be changing at every moment.

If you transform the path to `base_link`, that makes it easy to calculate this:
```cpp
distance_from_robot=hypot(ps.pose.position.x, ps.pose.position.y)
```
Also determining if the local goal is in front of the robot is just looking at the `x` component.

Both `transformGlobalPlan` and `transformPose` are utility functions that could be shared with dwb local planner, they are very similar.

Is ROS1 these utility functions were found in [local_planner_util](https://github.com/ros-planning/navigation/blob/db50577971b3489e2a693820b53aae198400c443/base_local_planner/src/local_planner_util.cpp) which controllers could reuse, which was very nice.